### ISSUE TITLE
Interactive code runner feature

### ISSUE THREAD

**Issue by convers39** [wsdjeg/SpaceVim#4194]
**Is your feature request related to a problem? Please describe.**

When I try to run a code that needs to collect user input, the code will hang on and stop running:

![image](https://user-images.githubusercontent.com/42597395/114296173-a700ca80-9ae4-11eb-94bd-d3d7644be66e.png)

The similar open issue here:
https://github.com/SpaceVim/SpaceVim/issues/3904#issue-722563547

It seems it is possible to customize a task to achieve that but I have no idea about the walkaround...

**Describe the solution you'd like**

So basically what I am thinking is that it will be great to have an interactive runner by default or at least can be set easily in the configuration file.

**Describe alternatives you've considered**

Or if there is a more elaborated instruction on how to custom the tasks to achieve this would be great too.

Much Much Much appreciated!

**Comment by wsdjeg** [comment]
I think it is same as https://github.com/SpaceVim/SpaceVim/issues/4128

**Comment by JuanQuiro** [comment]
hello, I get the same error when I run the code

At first the code runs strangely
![error1](https://user-images.githubusercontent.com/54697159/119071304-8e26e700-b9b7-11eb-938f-7becb9b0110c.jpg)

If I put an input, the process freezes. (cin>>a)
![error2](https://user-images.githubusercontent.com/54697159/119072119-fde9a180-b9b8-11eb-8dc4-d9053f071d91.jpg)

When I exit spacevim, I get the following error message
![error3](https://user-images.githubusercontent.com/54697159/119072211-1eb1f700-b9b9-11eb-82f8-4ca7b76006fa.jpg)

**Comment by wsdjeg** [comment]
this is feature of c programming language, you should use fflush() to update the buffer. see https://github.com/janet-lang/janet/issues/278, https://github.com/TypeStrong/ts-node/releases/tag/v8.9.1

**Comment by wsdjeg** [comment]
you can test with following c program:

```c
#include <stdio.h>

int main(void) {
  int a, b;

  while (scanf("%d%d", &a, &b) != EOF) {
    printf("%d\n", a + b);
    fflush(stdout);
  };

  return 0;
}
```

**Comment by wsdjeg** [comment]
please update spacevim, the version is too old, `on_exit` function has been updated. and for c language, you need to use `fflush(stdout)`. as the code runner get the result from stdout of the runner command. and if you want to use your owrn runner, you can read: https://spacevim.org/documentation/#custom-runner
